volume_data_converter/erddap_query.py

```diff
@@ -0,0 +1,109 @@
+import os
+from erddapy import ERDDAP
+import json
+import typer
+from halo import Halo
+from pathlib import Path
+
+app = typer.Typer()
+
+configuration_file_path = os.path.join(
+    Path(__file__).absolute().parent, "config", "erddap_configuration.json"
+)
+
+
+@app.command()
+def erddap_query(
+    output_file_path: str = typer.Argument(
+        ..., help="Location where the resuilting .raw files will be saved"
+    ),
+    erddap_configuration_file: str = typer.Option(
+        configuration_file_path,
+        help=" json file with the erddap connection settings",
+    ),
+):
+    """
+    Downloads erddap data according to the settings in the erddap_configuration.json file. Downloads the data and saves it in a NetCDF file
+    Keyword arguments:
+    output_file_path -- Path where the NC file will be saved. It could be a directory or a path to a file with nc extension (.nc)
+    erddap_configuration_file (Optional)-- Path to the errdap_confguration file. By default uses the one installed on this package.
+
+    """
+
+    print(f"configuration_file_path {erddap_configuration_file}")
+    try:
+        conf_file = open(erddap_configuration_file, "r")
+    except OSError:
+        raise (f" Could not open/read file: {erddap_configuration_file} ")
+
+    with conf_file:
+        configuration = json.load(conf_file)
+
+    erddap_connection_object = configuration["erddap_connection"]
+
+    ## check the output folder exists
+    if os.path.isdir(output_file_path):
+        typer.echo(
+            f"Ouput folder path is a directory. File-name auto assigned to: {erddap_connection_object['dataset_id']}.nc"
+        )
+        output_file_path = os.path.join(
+            output_file_path, erddap_connection_object["dataset_id"] + ".nc"
+        )
+    else:
+        ## check the file name format and extension
+        folder_path, filename = os.path.split(output_file_path)
+        if not os.path.isdir(folder_path):
+            raise (
+                f" Could not find output path: {output_file_path}. Check first argument"
+            )
+        name, ext = os.path.splitext(filename)
+        if ext != ".nc":
+            ext = ".nc"
+        output_file_path = os.path.join(folder_path, name + ext)
```

Comment:
This if/else clause is pretty hard to follow. Some additional comments might be useful. If I understand it correctly....

- The `if` handles if the input is only a path to a directory and will create a default file.
- The `else` handles if the input is a path + filename.
- It is worth noting that the `else` only handles a filename with extension. If no extension the code will crash? 

Perhaps it would be more clear to have it structured as: 

1. `if` an existing directory 
2. `elif` a file with extension - You can check this by `'.' in output_file_path` 
3. `else` raise input error

Reply:
I left a comment also related to this. I would vote for not doing so many checks and letting the errors just surfacing naturally

Reply:
1. In the first `if` I am checking if the directory exists. Just adding a message indicating that a files will be created in that folder and its name. I think the prospect users of this app need feedback from the app telling them what is happening.
2.  What is wrong by using the built-in function `os.path.splitext` ?  If there no extension, the variable `ext` is empty and it will enter into the `if ext != ".nc"`  that adds the extension. It wont crash.
3. In the `elif ` ( file with extension case ), I still need to check if in the given path, the parent folder exists.

Reply:
Ah I see, you're right.